exonum-java-binding-core/rust/integration_tests/tests/node_context.rs

```diff
@@ -31,74 +30,20 @@ lazy_static! {
 }
 
 #[test]
-fn submit_valid_transaction() {
-    let jclass = JObject::null().into();
+//TODO
+#[ignore]
+fn submit_transaction() {
     let (mut node, app_rx) = create_node();
-    let node_handle_guard = as_handle(&mut node);
-    let node_handle = node_handle_guard.get();
-    let (java_transaction, _raw_message) = create_mock_transaction(&EXECUTOR, true);
-    let marker_raw = RawMessage::from_vec(vec![1, 2, 3]);
-    let raw_message = marker_raw.clone();
-    unwrap_jni(EXECUTOR.with_attached(move |env: &JNIEnv| {
-        let submit = || {
-            let message = message_from_raw(env, &raw_message)?;
-            Java_com_exonum_binding_service_NodeProxy_nativeSubmit(
-                env.clone(),
-                jclass,
-                node_handle,
-                *java_transaction.as_obj(),
-                *message,
-                0,
-                raw_message.len() as i32,
-            );
-            let exception: JObject = env.exception_occurred()?.into();
-            assert!(exception.is_null());
-            Ok(())
-        };
-        unwrap_jni_verbose(&env, submit());
-        Ok(())
-    }));
+    let raw_transaction =
+        RawTransaction::new(0, ServiceTransaction::from_raw_unchecked(0, vec![1, 2, 3]));
+    node.submit(raw_transaction.clone()).unwrap();
     let sent_message = app_rx.wait().next().unwrap().unwrap();
     match sent_message {
-        ExternalMessage::Transaction(sent) => assert_eq!(&marker_raw, sent.raw()),
+        ExternalMessage::Transaction(sent) => assert_eq!(&raw_transaction, sent.payload()),
```

Comment:
The naming gets quite confusing: A ServiceTransaction has payload, and a SignedMessage
has payload and these are distinct things. Do you think anything can be clarified in the core? If not, at least please extract `sent.payload` in a variable.

Reply:
I don't think it's a problem. Payload of SignedMessage is deserialized ServiceTransaction + some additional info. I can't see the reason why they should be the same thing.